### PR TITLE
Add chat.rocket.RocketChat electron desktop app

### PR DESCRIPTION
**Pull request by andrevmatos**
Initial commit, v. 2.11.0, based on org.signal.Signal flathub's repo.
Smaller versions of the icon are added as part of this repository. RocketChat package provides a 512x512 icon, but [docs states that 64x64 and 128x128 versions are required](https://github.com/flathub/flathub/wiki/App-Requirements#application-icons). If the 512px version would be enough, I can remove it.

### CODE REVIEW

**Comment by nedrichards** [comment]
Have you tried building this natively with the yarn tools? https://github.com/flatpak/flatpak-builder-tools

**Comment by andrevmatos** [comment]
Yes, the `flatpak-yarn-generator.py` fails with `TypeError: unhashable type: 'dict'` when running on the [RocketChat's yarn.lock](https://github.com/RocketChat/Rocket.Chat.Electron/blob/develop/yarn.lock).

**Comment by nedrichards** [comment]
bot, build chat.rocket.RocketChat

**Comment by andrevmatos** [comment]
bot, build chat.rocket.RocketChat

**Comment by andrevmatos** [comment]
Added flathub.json to limit arches to x86_64 and i386, the ones provided by official packages.

**Comment by nedrichards** [comment]
bot, build chat.rocket.RocketChat

File: chat.rocket.RocketChat.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                "tar xf data.tar.xz",
+                "rm -f control.tar.gz data.tar.xz debian-binary",
+                "cp -r usr/* opt/* /app",
+                "install -D rocketchat.64x64.png /app/share/icons/hicolor/64x64/apps/rocketchat.png",
+                "install -D rocketchat.128x128.png /app/share/icons/hicolor/128x128/apps/rocketchat.png",
+                "chmod -R a-s,go+rX,go-w /app/Rocket.Chat+",
+                "sed -i 's|Exec=.*|Exec=/app/Rocket.Chat+/rocketchat %U|' /app/share/applications/rocketchat.desktop",
+                "install -Dm644 chat.rocket.RocketChat.appdata.xml /app/share/appdata/chat.rocket.RocketChat.appdata.xml"
+            ],
+            "sources": [
```
Comment (barthalion) [inline chat.rocket.RocketChat.json]: Please add tarball with source code too (so it gets exported to Sources extension).

File: chat.rocket.RocketChat.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            "build-commands": [
+                "ar x rocketchat_*.deb",
+                "rm -f rocketchat_*.deb",
+                "tar xf data.tar.xz",
+                "rm -f control.tar.gz data.tar.xz debian-binary",
+                "cp -r usr/* opt/* /app",
+                "install -D rocketchat.64x64.png /app/share/icons/hicolor/64x64/apps/rocketchat.png",
+                "install -D rocketchat.128x128.png /app/share/icons/hicolor/128x128/apps/rocketchat.png",
+                "chmod -R a-s,go+rX,go-w /app/Rocket.Chat+",
+                "sed -i 's|Exec=.*|Exec=/app/Rocket.Chat+/rocketchat %U|' /app/share/applications/rocketchat.desktop",
```
Comment (barthalion) [inline chat.rocket.RocketChat.json]: `desktop-file-edit`, can be fixed after merge.

**Comment by barthalion** [comment]
Hi @engelgabriel, would you be interested in having write access to the repository? Just in case, even if you are not interested in making it official.

**Comment by andrevmatos** [comment]
Added the source package (please, confirm if the parameters are correct), and replaced `sed` call with `desktop-file-edit`.

**Comment by barthalion** [comment]
bot, build chat.rocket.RocketChat

File: chat.rocket.RocketChat.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    <content_attribute id="language-discrimination">none</content_attribute>
+    <content_attribute id="social-chat">intense</content_attribute>
+    <content_attribute id="social-info">none</content_attribute>
+    <content_attribute id="social-audio">intense</content_attribute>
+    <content_attribute id="social-location">none</content_attribute>
+    <content_attribute id="social-contacts">intense</content_attribute>
+    <content_attribute id="money-purchasing">none</content_attribute>
+    <content_attribute id="money-gambling">none</content_attribute>
+  </content_rating>
+  <update_contact>andre.vmatos@gmail.com</update_contact>
```
[the remark this answers is not in the document]
Comment (andrevmatos) [reply chat.rocket.RocketChat.appdata.xml]: Done.

**Comment by barthalion** [comment]
bot, build chat.rocket.RocketChat

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/chat.rocket.RocketChat

Make sure to accept the invitation to get write access. Thanks!

**Comment by geekgonecrazy** [comment]
@andrevmatos this is awesome!  This flew totally under our radar.  Creating a flatpak is actually something we wanted to do and I just created an issue a couple of weeks ago: https://github.com/RocketChat/Rocket.Chat.Electron/issues/770

@barthalion We would definitely be interested in getting write access to help maintain this and maybe help set this up to build natively instead of from a .deb

**Pull request closed without merging**